### PR TITLE
chore: bump version for 0.4.0-rc.1 and changelogs

### PR DESCRIPTION
**Pull request by ragingsquirrel3**
Bumps version to 0.4.0-rc.1 and updates changelogs accordingly.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_flutter/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 0.4.0-rc.1 (2022-02-08)
+
+- fix(amplify_flutter): updates for latest flutter and dart versions #1333
+- feat(amplify_flutter): allow customers to override AmplifyClass methods #1325
+- fix(flutter): Password settings parsing #1323
```
Comment (HuiSF) [inline packages/amplify_flutter/CHANGELOG.md:5]: nit:
```suggestion
- fix(flutter): Password settings parsing #1323

```

**Review by HuiSF — APPROVED** [review]

**Review by haverchuck — APPROVED** [review]

File: packages/amplify_api/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,12 @@
+## 0.4.0-rc.1 (2022-02-08)
+
+- chore: bump Kotlin version to 1.6.10 (#1346)
+- feat(amplify_api): model-based Graphql helpers (#1211)
+
+### Breaking Changes
+
+- The `data` field is now nullable in `GraphQLResponse`.
+
```
Comment (Jordan-Nelson) [inline packages/amplify_api/CHANGELOG.md:9]:
```suggestion
- If your app has a dependency on Kotlin, the value of `ext.kotlin_version` set in `android/build.gradle` must be 1.5.31 or greater (1.6.10 recommended)

```

File: packages/amplify_auth_cognito/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 0.4.0-rc.1 (2022-02-08)
+
+- chore: bump Kotlin version to 1.6.10 (#1346)
```
Comment (Jordan-Nelson) [inline packages/amplify_auth_cognito/CHANGELOG.md:3]:
```suggestion
- chore: bump Kotlin version to 1.6.10 (#1346)

### Breaking Changes

- If your app has a dependency on Kotlin, the value of `ext.kotlin_version` set in `android/build.gradle` must be 1.5.31 or greater (1.6.10 recommended)
```

File: packages/amplify_core/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 0.4.0-rc.1 (2022-02-08)
+
+- chore: Many model-related type definitions moved from amplify_datastore_plugin_interface to amplify_core.
+
```
Comment (Jordan-Nelson) [inline packages/amplify_core/CHANGELOG.md:4]:
```suggestion
- chore: bump Kotlin version to 1.6.10 (#1346)

### Breaking Changes

- If your app has a dependency on Kotlin, the value of `ext.kotlin_version` set in `android/build.gradle` must be 1.5.31 or greater (1.6.10 recommended)

```

File: packages/amplify_datastore/CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,11 @@
+## 0.4.0-rc.1 (2022-02-08)
+
+- chore: bump Kotlin version to 1.6.10 (#1346)
+- chore(datastore): Export hub event types (#1330)
+- fix(datastore): Hub memory usage (#1201) 
+- feat(datastore): Add QueryPredicate.all (#1310)
+- chore(datastore): update integration tests schema (#1308)
+
```
Comment (Jordan-Nelson) [inline packages/amplify_datastore/CHANGELOG.md:8]:
```suggestion

### Breaking Changes

- If your app has a dependency on Kotlin, the value of `ext.kotlin_version` set in `android/build.gradle` must be 1.5.31 or greater (1.6.10 recommended)
```

File: packages/amplify_flutter/CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,9 @@
...
+
```
Comment (Jordan-Nelson) [inline packages/amplify_flutter/CHANGELOG.md:6]:
```suggestion
- chore: bump Kotlin version to 1.6.10 (#1346)

### Breaking Changes

- If your app has a dependency on Kotlin, the value of `ext.kotlin_version` set in `android/build.gradle` must be 1.5.31 or greater (1.6.10 recommended)

```

File: packages/amplify_storage_s3/CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,7 @@
+## 0.4.0-rc.1 (2022-02-08)
+
+- chore: bump Kotlin version to 1.6.10 (#1346)
```
Comment (Jordan-Nelson) [inline packages/amplify_storage_s3/CHANGELOG.md:3]:
```suggestion
- chore: bump Kotlin version to 1.6.10 (#1346)

### Breaking Changes

- If your app has a dependency on Kotlin, the value of `ext.kotlin_version` set in `android/build.gradle` must be 1.5.31 or greater (1.6.10 recommended)
```

File: packages/amplify_datastore/CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,14 @@
...
+### Breaking Changes
+
+- If your app has a dependency on Kotlin, the value of `ext.kotlin_version` set in `android/build.gradle` must be 1.5.31 or greater (1.6.10 recommended)
```
Comment (Jordan-Nelson) [inline packages/amplify_datastore/CHANGELOG.md:11]: Nit: Sorry, the original suggestion was probably missing a new line
```suggestion
- If your app has a dependency on Kotlin, the value of `ext.kotlin_version` set in `android/build.gradle` must be 1.5.31 or greater (1.6.10 recommended)

```

**Pull request merged**